src/flamegraph/merge.rs

```diff
@@ -50,31 +58,50 @@ fn flow<'a, LI, TI>(
         };
 
         //eprintln!("at {} ending frame {:?}", time, key);
-        let start_time = tmp.remove(&key).unwrap_or_else(|| {
+        let frame_time = tmp.remove(&key).unwrap_or_else(|| {
             unreachable!("did not have start time for {:?}", key);
         });
 
-        let key = TimedFrame {
+        let frame = TimedFrame {
             location: key,
-            start_time,
+            start_time: frame_time.start_time,
             end_time: time,
+            delta: frame_time.delta,
         };
-        frames.push(key);
+        frames.push(frame);
     }
 
-    for (i, func) in this.enumerate() {
+    let mut i = 0;
+    while this.peek().is_some() {
+        let func = this.next().unwrap();
         let key = Frame {
             function: func,
             depth: shared_depth + i,
         };
+
+        let is_last = this.peek().is_none();
+        let delta = match delta {
+            Some(_) if !is_last => Some(0),
+            d => d,
+        };
+        let frame_time = FrameTime {
+            start_time: time,
+            delta,
```

Comment:
I find it a little weird that the Perl version [does a `+=` for `delta`](https://github.com/brendangregg/FlameGraph/blob/1b1c6deede9c33c5134c920bdb7a44cc5528e9a7/flamegraph.pl#L588)...

Reply:
I also thought that was strange, so I did what made sense to me.

Reply:
Maybe leave a comment with a link to the Perl version?